lang/syn/src/parser/accounts/constraints.rs

```diff
@@ -589,6 +582,31 @@ impl<'ty> ConstraintGroupBuilder<'ty> {
             }
             _ => None,
         };
+        let spl_account = match (&token_mint, &token_authority) {
+            (None, None) => None,
+            _ => Some(ConstraintSplTokenAccount {
+                mint: token_mint.as_ref().map(|a| a.clone().into_inner().mint),
+                auth: token_authority
+                    .as_ref()
+                    .map(|a| a.clone().into_inner().auth),
+            }),
+        };
```

Comment:
no need for a match here, is there?

same with the `spl_mint = ...` below

Reply:
Hmm, I think it should be there.
Have you got an idea to improve this?

Reply:
yea not sure what I was thinking here. looks correct to me now!